libnetwork/drivers/bridge/port_mapping_linux.go

```
@@ -764,22 +752,30 @@ func (n *bridgeNetwork) releasePortBindings(pbs []portBinding) error {
 }
 
 func (n *bridgeNetwork) setPerPortIptables(b portBinding, enable bool) error {
-	if (b.IP.To4() != nil) != (b.HostIP.To4() != nil) {
-		// The binding is between containerV4 and hostV6 (not vice-versa as that
-		// will have been rejected earlier). It's handled by docker-proxy, so no
-		// additional iptables rules are required.
-		return nil
-	}
 	v := iptables.IPv4
 	if b.IP.To4() == nil {
 		v = iptables.IPv6
 	}
-
 	if enabled, err := n.iptablesEnabled(v); err != nil || !enabled {
```

Not new, but this is the only location where `n.iptablesEnabled()` is called, and we're ignoring the error; should the error be returned, so that it can be logged (where relevant)?

---

s/ignoring/using it as boolean/

---

That was quick! Thank you.

I think the error's basically impossible ... no `driver`, for a `bridgeNetwork` that's hanging off the `driver`.

I've got some pretty major refactoring coming up (related to splitting iptables rules out of the bridge driver, to make way for nftables)... it'll get rid of a lot of this sort of logic -  ok to leave it until then?

---

Definitely! I also peeked at the error, and concluded roughly the same. My eye just caught it, and the combination of "and nobody else is using this?" 😂